packages/react/src/menu/trigger/MenuTrigger.tsx

```diff
@@ -199,6 +367,29 @@ export type MenuTriggerState = {
 };
 
 export namespace MenuTrigger {
-  export type Props = MenuTriggerProps;
+  export type Props<Payload = unknown> = MenuTriggerProps<Payload>;
   export type State = MenuTriggerState;
 }
+
+/**
+ * Determines whether to ignore clicks after a hover-open.
+ */
```

Comment:
Popover also uses this, maybe it could be extracted as a common util

Reply:
I'm planning a follow-up PR to create common utils, split other Floating UI interaction hooks and generally clean up the code.